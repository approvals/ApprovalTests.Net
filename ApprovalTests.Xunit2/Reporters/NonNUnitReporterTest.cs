namespace ApprovalTests.Xunit2.Reporters
{
    using ApprovalTests.Reporters;

    using ApprovalUtilities.Utilities;

    using Xunit;

    public class NonNUnitReporterTest
    {
        [Fact]
        public void TestNunitIsNotWorkingFromXUnit()
        {
            Approvals.SetCaller();
            Assert.False(NUnitReporter.INSTANCE.IsWorkingInThisEnvironment("default.txt"));
        }

        [Fact]
        public void TestXunitIsWorking()
        {
            Approvals.SetCaller();
            Assert.True(XUnitReporter.INSTANCE.IsWorkingInThisEnvironment("default.txt"));
        }

        [Fact]
        public void TestXunitReporterIsWorking()
        {
            var e = ExceptionUtilities.GetException(() => XUnit2Reporter.INSTANCE.AssertEqual("Hello", "Hello2"));
            Approvals.Verify(e.Message);
        }

        [Fact]
        public void TestXunitReporterIgnoreEndLines()
        {
            XUnit2Reporter.INSTANCE.ShouldIgnoreLineEndings = true;
            XUnit2Reporter.INSTANCE.AssertEqual("Hello\nHello", "Hello\r\nHello");
        }

        [Fact]
        public void TestXunitReporterIgnoreEndLinesAndFailAfter()
        {
            XUnit2Reporter.INSTANCE.ShouldIgnoreLineEndings = true;
            var e = ExceptionUtilities.GetException(() => XUnit2Reporter.INSTANCE.AssertEqual("Hello\nHello", "Hello\r\nHello2"));
            Approvals.Verify(e.Message);
        }
    }
}