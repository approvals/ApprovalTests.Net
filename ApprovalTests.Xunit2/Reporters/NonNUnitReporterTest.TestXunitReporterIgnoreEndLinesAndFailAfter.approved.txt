Assert.Equal() Failure
                      ↓ (pos 11)
Expected: Hello\nHello
Actual:   Hello\r\nHello2
                        ↑ (pos 12)